pkg/console/install_panels.go

```diff
@@ -498,6 +546,350 @@ func addNetworkPanel(c *Console) error {
+			return err
+		}
+		if validateErrMsg != "" {
+			return c.setContentByName(networkValidatorPanel, validateErrMsg)
+		}
+		return showNext(c, gatewayPanel)
+	}
+	addressV.KeyBindings = map[gocui.Key]func(*gocui.Gui, *gocui.View) error{
+		gocui.KeyArrowUp: func(g *gocui.Gui, v *gocui.View) error {
+			validateErrMsg, err := validateAddress()
+			if err != nil {
+				return err
+			}
+			if validateErrMsg != "" {
+				return c.setContentByName(networkValidatorPanel, validateErrMsg)
+			}
+			return showNext(c, hostNamePanel)
+		},
+		gocui.KeyArrowDown: addressVConfirm,
+		gocui.KeyEnter:     addressVConfirm,
+		gocui.KeyEsc:       goBack,
+	}
+	setLocation(addressV.Panel, 3)
+	c.AddElement(addressPanel, addressV)
+
+	// gatewayV
+	gatewayV.PreShow = func() error {
+		c.Gui.Cursor = true
+		gatewayV.DefaultValue = installData.Gateway
+		return c.setContentByName(titlePanel, "Configure Gateway")
+	}
+	validateGateway := func() (string, error) {
+		gateway, err := gatewayV.GetData()
+		if err != nil {
+			return "", err
+		}
+		if err = validateIP(gateway); err != nil {
+			return err.Error(), nil
+		}
+		validatorV.Close()
+		installData.Gateway = gateway
+		return "", nil
+	}
+	gatewayVConfirm := func(g *gocui.Gui, v *gocui.View) error {
+		validateErrMsg, err := validateGateway()
+		if err != nil {
+			return err
+		}
+		if validateErrMsg != "" {
+			return c.setContentByName(networkValidatorPanel, validateErrMsg)
+		}
+		return showNext(c, dnsServersPanel)
+	}
+	gatewayV.KeyBindings = map[gocui.Key]func(*gocui.Gui, *gocui.View) error{
+		gocui.KeyArrowUp: func(g *gocui.Gui, v *gocui.View) error {
+			validateErrMsg, err := validateGateway()
+			if err != nil {
+				return err
+			}
+			if validateErrMsg != "" {
+				return c.setContentByName(networkValidatorPanel, validateErrMsg)
+			}
+			return showNext(c, addressPanel)
+		},
+		gocui.KeyArrowDown: gatewayVConfirm,
+		gocui.KeyEnter:     gatewayVConfirm,
+		gocui.KeyEsc:       goBack,
+	}
+	setLocation(gatewayV.Panel, 3)
+	c.AddElement(gatewayPanel, gatewayV)
+
+	// dnsServersV
+	dnsServersV.PreShow = func() error {
+		c.Gui.Cursor = true
+		if installData.DNSServers != "" {
+			dnsServersV.DefaultValue = installData.DNSServers
+		}
+		return c.setContentByName(titlePanel, "Configure DNS Servers")
+	}
+	validateDNSServers := func() (string, error) {
+		dnsServers, err := dnsServersV.GetData()
+		if err != nil {
+			return "", err
+		}
+		dnsServerList := strings.Split(dnsServers, ",")
+		for _, dnsServer := range dnsServerList {
+			if err = validateIP(dnsServer); err != nil {
+				return err.Error(), nil
+			}
+		}
+		installData.DNSServers = dnsServers
+		cfg.Config.K3OS.DNSNameservers = dnsServerList
+
+		// run-cmd
+		configureNetworkManual := fmt.Sprintf("/sbin/configure-network-manual %s %s %s %s %s",
+			installData.Interface, installData.IP, installData.NetMask, installData.Gateway, strings.Join(dnsServerList, " "))
+		cfg.Config.Runcmd = []string{configureNetworkManual}
```

Comment:
Does the static IP take effect only after installation?
If that's the case, the fetching URL behavior in SSH keys and Cloud Config dialog might not work.

Reply:
During installation，live os will use dhcp to fetch URL.

Reply:
Some users request to install with no DHCP. Sounds like this does not satisfy their requirements